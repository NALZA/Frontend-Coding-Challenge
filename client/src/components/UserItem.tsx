import styled from 'styled-components';

const Card = styled.div`
    display: flex;
    justify-content: flex-start;
    align-items: center;
    gap: 20px;
    height: 60px;
`;

const Avatar = styled.h3`
    display: flex;
    align-items: center;
    justify-content: center;
    color: SteelBlue;
    background-color: #e4f3ff;
    border-radius: 10px;
    width: 50px;
    height: 50px;
`;

const NameTag = styled.div`
    display: flex;
    flex-direction: column;
    gap: 5px;
`;

const Role = styled.p`
    margin: 0;
    font-size: .95em;
    color: gray;
`;

const Name = styled.p`
    margin: 0;
    font-size: 1.17em;
`;

interface Props {
    name: string,
    role: string
}

export const UserItem = (props: Props) => {
    return (
        <Card>
            <Avatar data-testid="avatar">
                {props.name[0]}
            </Avatar>
            <NameTag>
                <Name data-testid="name">{props.name}</Name>
                <Role data-testid="role">{props.role}</Role>
            </NameTag>
        </Card>
    )
}
