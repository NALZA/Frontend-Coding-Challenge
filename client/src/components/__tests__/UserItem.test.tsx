import { render} from '@testing-library/react';
import { UserItem } from '../UserItem';

describe("Creates a card with a user avatar, name and role", () => {
    it("Creates a manager card for Jim James", () =>{
        const card = render(<UserItem name="Jim James" role="Manager" /> );
        expect(card.getByTestId('avatar').textContent).toBe('J');
        expect(card.getByTestId('name').textContent).toBe('Jim James');
        expect(card.getByTestId('role').textContent).toBe('Manager');
    })
})