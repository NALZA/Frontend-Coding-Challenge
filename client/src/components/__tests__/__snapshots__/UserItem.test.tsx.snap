// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a card with a user avatar, name and role Tests snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM khowmD"
      >
        <h3
          class="sc-gtsrHT lanIWJ"
          data-testid="avatar"
        >
          J
        </h3>
        <div
          class="sc-dlnjwi jMwFjE"
        >
          <p
            class="sc-eCApnc ignkqQ"
            data-testid="name"
          >
            Jim James
          </p>
          <p
            class="sc-hKFxyN jvCncI"
            data-testid="role"
          >
            Manager
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM khowmD"
    >
      <h3
        class="sc-gtsrHT lanIWJ"
        data-testid="avatar"
      >
        J
      </h3>
      <div
        class="sc-dlnjwi jMwFjE"
      >
        <p
          class="sc-eCApnc ignkqQ"
          data-testid="name"
        >
          Jim James
        </p>
        <p
          class="sc-hKFxyN jvCncI"
          data-testid="role"
        >
          Manager
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
